src/plugins/discover/public/utils/get_type_for_field_icon.ts

```diff
@@ -13,5 +13,10 @@ import { DataViewField } from '@kbn/data-views-plugin/common';
  *
  * We define custom logic for Discover in order to distinguish between various "string" types.
  */
-export const getTypeForFieldIcon = (field: DataViewField) =>
-  field.type === 'string' && field.esTypes ? field.esTypes[0] : field.type;
+export const getTypeForFieldIcon = (field: DataViewField) => {
+  const esType = field.esTypes?.[0] || null;
+  if (esType && ['_id', '_index'].includes(esType)) {
```

Comment:
this little change is 🥳 . Finally no more unknown icons for `_id` and `_index`

<img width="286" alt="Bildschirmfoto 2022-11-09 um 17 15 35" src="https://user-images.githubusercontent.com/463851/200882928-be2d1b44-ac66-4484-859f-b29aaad3041e.png">